src/editorconfig.rs

```diff
@@ -0,0 +1,374 @@
+use crate::{CallParenType, CollapseSimpleStatement, Config, IndentType, LineEndings, QuoteStyle};
+use ec4rs::{
+    properties_of,
+    property::{EndOfLine, IndentSize, IndentStyle, MaxLineLen, TabWidth, UnknownValueError},
+    rawvalue::RawValue,
+    Error, Properties, PropertyKey, PropertyValue,
+};
+use std::path::Path;
+
+// Read the EditorConfig files that would apply to a file at the given path
+pub fn parse(config: Option<Config>, path: &Path) -> Result<Option<Config>, Error> {
```

Comment:
Is it necessary for the input and output here to be `Option<Config>`? It looks like `None` is only input in the stdin version, but can't we just pass the base `config` the same as the filepath version

Reply:
It was used for logging if no properties were found, this is removed now.